sdk/core/src/CommonNetworkMember/BaseNetworkMember/AnchoringService.ts

```diff
@@ -0,0 +1,80 @@
+import { ParsedOptions } from '../../shared/getOptionsFromEnvironment'
+import { extractSDKVersion } from '../../_helpers'
+import { ELEM_DID_METHOD, JOLO_DID_METHOD } from '../../_defaultConfig'
+import { RegistryApiService } from '@affinidi/internal-api-clients'
+import { KeysService } from '@affinidi/common'
+
+export class AnchoringService {
+  constructor(
+    private readonly encryptedSeed: string,
+    private readonly password: string,
+    private readonly didDocument: any,
+    private readonly nonce: number,
+    private readonly options: ParsedOptions,
+    private readonly anchoredDidElem?: boolean,
+  ) {}
+
+  private get api() {
+    return new RegistryApiService({
+      registryUrl: this.options.basicOptions.registryUrl,
+      accessApiKey: this.options.accessApiKey,
+      sdkVersion: extractSDKVersion(),
+    })
+  }
+
+  private async anchorDidWithJoloMethod(keysService: KeysService, seedHex: string) {
+    const did = this.didDocument.id
+    const signedDidDocument = await keysService.signDidDocument(this.didDocument)
+
+    const { body: bodyDidDocument } = await this.api.putDocumentInIpfs({ document: signedDidDocument })
+    const didDocumentAddress = bodyDidDocument.hash
+
+    const {
+      body: { digestHex },
+    } = await this.api.createAnchorTransaction({ nonce: this.nonce, did, didDocumentAddress })
+
+    const transactionSignatureJson =
+      digestHex && digestHex !== '' ? await keysService.createTransactionSignature(digestHex, seedHex) : ''
+
+    const transactionPublicKey = KeysService.getAnchorTransactionPublicKey(seedHex, JOLO_DID_METHOD)
+    const ethereumPublicKeyHex = transactionPublicKey.toString('hex')
+
+    await this.api.anchorDid({
+      did,
+      didDocumentAddress,
+      ethereumPublicKeyHex,
+      transactionSignatureJson,
+      nonce: this.nonce,
+    })
+  }
+
+  private async anchorDidWithElemMethod() {
+    const did = this.didDocument.id
+    try {
+      await this.api.anchorDid({
+        did,
+        didDocumentAddress: '',
+        ethereumPublicKeyHex: '',
+        transactionSignatureJson: '',
```

Comment:
Why do we rely on empty strings instead of e.g. null or undefined?

Reply:
don't know

Reply:
![image](https://user-images.githubusercontent.com/36485067/134004885-0286fabf-b727-4a8f-92a8-2873eff939ec.png)